### PR TITLE
feat(erc20): add indexes at genesis state to track active token pairs

### PR DESCRIPTION
**Pull request by zsystm**
### Naming Change
The following maps exist for the purpose of indexing to find `TokenPair`. I believe the new names better reflect their purpose:
- `DenomMap` --> `TokenPairDenomIndex`
- `ERC20Map` --> `TokenPairERC20AddressIndex`

Aligned the following method names with their original intent and actual actions:
- `GetDenomMap` --> `GetTokenPairIdByDenom`
- `GetERC20Map` --> `GetTokenPairIdByERC20Addr`

### Added indexes as state

**Problem**
- [The original InitGenesis logic](https://github.com/Canto-Network/Canto/blob/v7.0.1/x/erc20/genesis.go#L26-L31) iterates over all existing TokenPair to set denomIndex(= `denomMap`) and erc20AddressIndex(= `erc20Map`).
- The problem arises when multiple token pairs exist for same denom. The iteration order is based on bytes, not the actual registration time.
- This means the logic might not correctly identify the most recently registered TokenPair.
- For example, the denomIndex for `ibc/13B6057538B93225F6EBACCB64574C49B2C1568C5AE6CCFE0A039D7DAC02BF29` ends up with the TokenPair associated with `0x2C68D1d6aB986Ff4640b51e1F14C716a076E44C4`, which is incorrect.
- The correct TokenPair should be the one associated with `0xD32eB974468ed767338533842D2D4Cc90B9BAb46`, as it was registered later and is currently active.

**Solution**
- Added indexes at genesis state to track active token pair. Now we can track the latest(= active) one if there were multiple token pairs at the same denom.

### Notes
- Actual logic fix will be handled as another PR.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted the correct branch (see [PR Targeting](https://github.com/canto-network/canto/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] included the necessary unit and integration tests
- [x] reviewed "Files changed" and left comments if necessary
 
### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/erc20/genesis_test.go [hunk 1]
```diff
@@ -82,30 +129,35 @@ func (suite *GenesisTestSuite) TestERC20InitGenesis() {
...
+				denomIdxs,
+				erc20AddrIdxs,
+			),
 		},
 	}
 
 	for _, tc := range testCases {
-
+		// TODO: fix duplicate problem and uncomment this line
+		//suite.Nil(tc.genesisState.Validate(), "genesis state should be valid")
```
Comment (dudong2) [inline x/erc20/genesis_test.go]: no need to uncomment?

Comment (zsystm) [reply x/erc20/genesis_test.go]: @dudong2 
I removed it. Thanks.

**Comment by tkkwon1998** [comment]
lgtm!

File: x/erc20/genesis_test.go [hunk 2]
```diff
@@ -21,6 +22,66 @@ import (
...
+		customERC20,
+	}
+	denomIdxs = []types.TokenPairDenomIndex{
+		{
+			Denom:       customERC20.Denom,
+			TokenPairId: customERC20.GetID(),
+		},
+		{
+			Denom: uqstars,
+			// denomIdx must have latest token pair id
```
Comment (dongsam) [inline x/erc20/genesis_test.go]: The token pair id is a hash, not a number, but currently, "latest" might be misinterpreted as numerically largest. How about modifying it as suggested below?

```suggestion
			// denomIdx must have the latest token pair id assigned
```

**Review by dongsam — APPROVED** [review]
LGTM, apart from the chore suggestion on the comment

**Review by dudong2 — APPROVED** [review]
lgtm

**Pull request merged**